File: Comparasion_between_wxWidgets_and_wxPython/A_simple_menu_window.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
#-----------------------------------------------------------------------------
# Purpose: a simple menu window, with wxPython.
# Date: Wed 01/25/2017
#-----------------------------------------------------------------------------

import wx

class Frame(wx.Frame):

    def __init__(self, title, pos, size):
        wx.Frame.__init__(self, None, -1, title, pos, size)

        menuFile = wx.Menu()
        menuFile.Append(1, "&About...")
        menuFile.AppendSeparator()
        menuFile.Append(2, "E&xit")
        
        menuBar = wx.MenuBar()
        menuBar.Append(menuFile, "&File")
        self.SetMenuBar(menuBar)

        self.CreateStatusBar()
        self.SetStatusText("Welcome to wxPython!")

        self.Bind(wx.EVT_MENU, self.OnAbout, id=1)
        self.Bind(wx.EVT_MENU, self.OnQuit, id=2)

    def OnQuit(self, event):
        self.Close()

    def OnAbout(self, event):
        wx.MessageBox("This is a wxPython Hello world sample", 
                "About Hello World", wx.OK | wx.ICON_INFORMATION, self)

class App(wx.App):

    def OnInit(self):
        frame = Frame("Hello World", (50, 60), (450, 340))
        frame.Show()
        self.SetTopWindow(frame)
        return True

if __name__ == '__main__':
    app = App()
    app.MainLoop()

File: Comparasion_between_wxWidgets_and_wxPython/wxWidgets/A_simple_menu_window/A_simple_menu_window.cpp
//----------------------------------------------------------------------------
// Purpose: a simple menu window, using wxWidgets.
// Date: 2017-01-25
//----------------------------------------------------------------------------

#include <wx/wx.h>

class App : public wxApp
{
    virtual bool OnInit();
};

class Frame : public wxFrame
{
public:
    Frame(const wxString & title, const wxPoint & pos, const wxSize & size);
    void OnQuit(wxCommandEvent & event);
    void OnAbout(wxCommandEvent & event);
    DECLARE_EVENT_TABLE()
};

enum
{
    ID_Quit = 1,
    ID_About,
};

BEGIN_EVENT_TABLE(Frame, wxFrame)
    EVT_MENU(ID_Quit, Frame::OnQuit)
    EVT_MENU(ID_About, Frame::OnAbout)
END_EVENT_TABLE()

IMPLEMENT_APP(App)

bool App::OnInit()
{
    Frame *frame = new Frame("Hello World", wxPoint(50, 50), wxSize(450, 340));
    frame->Show(TRUE);
    SetTopWindow(frame);
    return TRUE;
}

Frame::Frame(const wxString & title, const wxPoint & pos, const wxSize & size):
    wxFrame((wxFrame *)NULL, -1, title, pos, size)
{
    wxMenu *menuFile = new wxMenu;
    menuFile->Append(ID_About, "&About...");
    menuFile->AppendSeparator();
    menuFile->Append(ID_Quit, "E&xit");

    wxMenuBar *menuBar = new wxMenuBar;
    menuBar->Append(menuFile, "&File");
    SetMenuBar(menuBar);

    CreateStatusBar();
    SetStatusText("Welcome to wxWidgets!");
}

void Frame::OnQuit(wxCommandEvent & WXUNUSED(event))
{
    Close(TRUE);
}

void Frame::OnAbout(wxCommandEvent & WXUNUSED(event))
{
    wxMessageBox("This is a wxWidgets Hello world sample", "About Hello World",
                 wxOK | wxICON_INFORMATION, this);
}

